tests/test_other.py

```diff
@@ -3595,6 +3595,10 @@ def test_dashM_respect_dashO(self):
     assert len(with_dash_o) == 0
     assert len(without_dash_o) != 0
 
+  def test_null_device_as_input(self):
+    out = Popen([PYTHON, EMXX, '-E', '-dM', '-xc', '/dev/null'], stdout=PIPE).communicate()[0]
```

Comment:
You are right, however, I have no idea how the Emscripten's unit testing is being carried out on Windows. The test scripts appear to be *nix-centric to me. At the very least, this test ensures that the changes in this PR does not cause any regression issue on *nix platforms.

If the test suite could be run on Windows then we could add another similar test as above but for Windows by simply replacing "/dev/null" to "nul".